db/memtable_list.cc

```diff
@@ -332,6 +333,8 @@ Status MemTableList::InstallMemtableFlushResults(
   mu->AssertHeld();
 
   // flush was successful
+  // Record the status on the memtable object. Either this call or a future call
```

Comment:
Not necessarily a **future** call. It can be read by either a thread which has already started flushing older memtables, or future threads.

Reply:
but the other thread must be waiting behind the mutex otherwise it has done the manifest recording already. if it is waiting behind mutex, then it will call this function afterward, ergo the future call.

Reply:
Is it possible that a previous thread has already set `commit_in_progress_` to true, and releases the db mutex while calling `LogAndApply`? At this time, the current thread can have the mutex, flush the memtables, and enters the function of `InstallMemtableFlushResults`. It sees that `commit_in_progress_==true`, thus just returns. Later, the previous thread will discover the newly flushed memtables and commit them to MANIFEST.